apis/gateway/v1beta2/apirule_types.go

```diff
@@ -0,0 +1,236 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	"istio.io/api/networking/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// StatusCode describing APIRule.
+type StatusCode string
+
+const (
+	//StatusOK is set when the reconciliation finished successfully
+	StatusOK StatusCode = "OK"
+	//StatusSkipped is set when reconciliation of the APIRule component was skipped
+	StatusSkipped StatusCode = "SKIPPED"
+	//StatusError is set when an error happened during reconciliation of the APIRule
+	StatusError StatusCode = "ERROR"
+	//StatusWarning is set if a user action is required
+	StatusWarning StatusCode = "WARNING"
+)
+
+// APIRuleSpec defines the desired state of ApiRule.
+type APIRuleSpec struct {
+	// Specifies the URL of the exposed service.
+	// +kubebuilder:validation:MinItems=1
+	Hosts []*string `json:"hosts"`
+	// Describes the service to expose.
+	// +optional
+	Service *Service `json:"service,omitempty"`
+	// Specifies the Istio Gateway to be used.
+	// +kubebuilder:validation:Pattern=`^[0-9a-z-_]+(\/[0-9a-z-_]+|(\.[0-9a-z-_]+)*)$`
+	Gateway *string `json:"gateway"`
+	// Specifies CORS headers configuration that will be sent downstream
+	// +optional
+	CorsPolicy *CorsPolicy `json:"corsPolicy,omitempty"`
+	// Represents the array of Oathkeeper access rules to be applied.
+	// +kubebuilder:validation:MinItems=1
+	Rules []Rule `json:"rules"`
+	// +optional
+	Timeout *Timeout `json:"timeout,omitempty"`
+}
+
+// APIRuleStatus describes the observed state of ApiRule.
+type APIRuleStatus struct {
```

Comment:
Do we want to have a follow up to discuss the status? I remember that we thought about changing this in the future.